### PR TITLE
perlPackages: bulk update

### PR DESCRIPTION
**Pull request by volth**
update `perlPackages.*`

successfully built on `x86_64-linux` and `aarch64-linux`

### CODE REVIEW

**Comment by volth** [comment]
`perlPackages.DBIxClass: 0.082841 -> 0.082842` broke `Catalyst-Authentication-Store-DBIx-Class`, so I am going to undo that commit

**Review by stigtsp — COMMENTED** [review]
All `perlPackages` build and test ok on 86491c53ef9544e3cda87f5abb35c59ceb7caad1

Some dependency cleanups do cause issues however. For example `CPANMini` which included `LWPProtocolHttps` so `minicpan` can use https mirrors. 

`minicpan -r https://www.cpan.org -l /tmp`

I'll go trough these issues in more detail later tonight.

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -3546,7 +3546,7 @@ let
       sha256 = "5a297afc3e367ad80811464d4eb7e4dd3caff8ba499cdd2b558f6279443a7657";
     };
     nativeBuildInputs = stdenv.lib.optional stdenv.isDarwin shortenPerlShebang;
-    propagatedBuildInputs = [ FileHomeDir LWP LWPProtocolHttps URI ];
+    propagatedBuildInputs = [ FileHomeDir LWP ];
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:3547]:
```suggestion
    propagatedBuildInputs = [ FileHomeDir LWP LWPProtocolHttps URI ];
```
So minicpan supports https enabled mirrors

**Review by stigtsp — CHANGES_REQUESTED** [review]
My assumption was wrong about being more issues with the cleanups, only found one issue with `CPANMini`.

File: pkgs/top-level/perl-packages.nix (hunk 1 above)
Comment (volth) [reply pkgs/top-level/perl-packages.nix]: done.
`LWP` and `URI` are not needed, `LWPProtocolHttps` brings them too

**Review by stigtsp — APPROVED** [review]
Looks good to me

**Comment by stigtsp** [comment]
@volth @aanderse @vcunat Any objections if I merge this to `staging`?

**Comment by volth** [comment]
@stigtsp do not forget https://github.com/NixOS/nixpkgs/pull/97522 which may be more important because of release deadline

**Pull request merged**